references/embedding/train.py

```diff
@@ -0,0 +1,149 @@
+    for data in loader:
+        samples, _labels = data[0].to(device), data[1]
+        out = model(samples)
+        embeds = torch.cat((embeds, out), dim=0) if embeds is not None else out
+        labels = torch.cat((labels, _labels), dim=0) if labels is not None else _labels
+
+    dists = torch.cdist(embeds, embeds)
+
+    labels = labels.unsqueeze(0)
+    targets = labels == labels.t()
+
+    mask = torch.ones(dists.size()).triu() - torch.eye(dists.size(0))
+    dists = dists[mask == 1]
+    targets = targets[mask == 1]
+
+    best_thresh = 0.01
+    best_correct = 0
+    for thresh in torch.arange(0.0, 1.51, 0.01):
+        predictions = dists <= thresh.to(device)
+        correct = torch.sum(predictions == targets.to(device)).item()
+        if correct > best_correct:
+            best_thresh = thresh
+            best_correct = correct
+
+    accuracy = 100.0 * best_correct / dists.size(0)
+    print('accuracy: {:.3f}%, threshold: {:.2f}'.format(accuracy, best_thresh))
+
+
+def save(model, epoch, save_dir, file_name):
+    file_name = 'epoch_' + str(epoch) + '__' + file_name
+    save_path = os.path.join(save_dir, file_name)
+    torch.save(model.state_dict(), save_path)
+
+
+def main(args):
+    device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+    p = args.labels_per_batch
+    k = args.samples_per_label
+    batch_size = p * k
+
+    model = EmbeddingNet()
+    if args.resume:
+        model.load_state_dict(torch.load(args.resume))
+
+    model.to(device)
+
+    criterion = TripletMarginLoss(margin=args.margin)
+    optimizer = Adam(model.parameters(), lr=args.lr)
+
+    transform = transforms.Compose([transforms.Lambda(lambda image: image.convert('RGB')),
+                                    transforms.Resize((224, 224)),
+                                    transforms.ToTensor()])
+
+    train_dataset = FashionMNIST(args.train_data, train=True, transform=transform, download=True)
+    test_dataset = FashionMNIST(args.test_data, train=False, transform=transform, download=True)
+
+    targets = train_dataset.targets.tolist()
+    train_loader = DataLoader(train_dataset, batch_size=batch_size,
+                              sampler=PKSampler(targets, p, k),
+                              num_workers=4)
+
+    for epoch in range(1, args.epochs + 1):
+        print('Training...')
+        train_epoch(model, optimizer, criterion, train_loader, device, epoch, args.print_freq)
+
+        print('Evaluating...')
+        evaluate(model, test_dataset, device)
+
+        print('Saving...')
+        save(model, epoch, args.save_dir, 'ckpt.pth')
+
+
+def parse_args():
+    import argparse
+    parser = argparse.ArgumentParser(description='PyTorch Embedding Learning')
+
+    parser.add_argument('--train-data', default='/tmp/pyemb/train/',
+                        help='FashionMNIST train dataset path')
+    parser.add_argument('--test-data', default='/tmp/pyemb/test/',
```

Comment:
At least for `FashionMNIST`, the root folder is the same, so this can probably be removed

Reply:
Changed the arg to `--dataset-dir`.